registry/server/templates/services/renderTemplate.ts

```diff
@@ -71,18 +65,87 @@ function matchIncludesAttributes(template: any) {
 
         return includes;
     }, {});
-
-    return includesAttributes;
 }
 
-function selectDuplicateIncludesAttributes(includesAttributes: any) {
+function selectDuplicateIncludesAttributes(includesAttributes: IncludesAttributes): Array<IncludeAttributes> {
+    const attributes = Object.values(includesAttributes);
+
+    if (!attributes.length) {
+        return [];
+    }
+
     return Object.values(includesAttributes).filter((
-        currentAttributes: any,
+        currentAttributes: IncludeAttributes,
         index,
         attributes
     ) =>
         (currentAttributes.id || currentAttributes.src) &&
-        attributes.findIndex(({id, src}: any) => id === currentAttributes.id || src === currentAttributes.src) !== index);
+        attributes.findIndex(({ id, src }: IncludeAttributes) => id === currentAttributes.id || src === currentAttributes.src) !== index);
+}
+
+async function fetchIncludes(includesAttributes: IncludesAttributes): Promise<Array<string>> {
+    const includes = Object.keys(includesAttributes);
+
+    if (!includes.length) {
+        return [];
+    }
+
+    return Promise.all(includes.map(async (include: string) => {
+        try {
+            const {
+                src,
+                timeout = 10000,
+                id,
+            } = includesAttributes[include];
+
+            if (!id || !src) {
+                throw new Error(`Necessary attribute src or id was not provided by ${include}!`);
+            }
+
+            const {
+                data,
+                headers: {
+                    link,
+                }
+            } = await axios.get(src, {
+                timeout: +timeout,
+            });
+
+            const stylesheets = selectStylesheets(link);
+
+            if (!stylesheets.length) {
+                return wrapWithComments(id, data);
+            }
+
+            return wrapWithComments(id, data + '\n' + stylesheets.join('\n'));
+        } catch (error) {
+            // TODO Need to add correct error handling
+            console.error(error);
+            return include;
+        }
+    }));
+}
+
+function selectStylesheets(link: string): Array<string> {
+    const includeLinkHeader = parseLinkHeader(link);
+
+    if (!includeLinkHeader.length) {
+        return [];
+    }
+
+    return includeLinkHeader.reduce((stylesheets: Array<string>, attributes: any) => {
+        if (attributes.rel !== 'stylesheet') {
+            return stylesheets;
+        }
+
+        stylesheets.push(`<link rel="stylesheet" href="${attributes.uri}">`);
+
+        return stylesheets;
+    }, []);
+}
+
+function wrapWithComments(id: string, data: string): string {
+    return `<!-- Include "${id}" START -->\n` + data + `\n<!-- Include "${id}" END -->`;
```

Comment:
Could you rename it to `Template include "${id}" START`?

Reply:
To clearly distinguish template includes from ILC ones.

Reply:
Sounds reasonable
Renamed